### PR TITLE
Fixing faceted search to return 404 if supplied field name does not e…

### PR DESCRIPTION
**Pull request by karlbaker02**
…xist for the given register

### CODE REVIEW

**Comment by chr1sdi** [comment]
tested and can be rebased, can you rebase before we merge please?

**Comment by michaelabenyohai** [comment]
Would it be possible to move `registerContainsField()` into `PostgresRegister` (we should rename this to something without Postgres in it). This is because it doesn't require any Postgres operation and therefore I feel like it does not need to be all the way down in `PostgresDriver`. 

It would also mean that we pass `RegistersConfiguration` to `PostgresRegister` instead of to `PostgresDriver`. This makes sense since we already pass `RegisterNameConfiguration` to `PostgresRegister`.

Please could you also create an interface such as `RegisterFieldsConfiguration` and pass only the required field data in - this makes it easier to Mock and test the `PostgresRegister` object.

File: src/main/java/uk/gov/register/core/PostgresRegister.java [hunk 1]
```diff
@@ -1,7 +1,11 @@
 package uk.gov.register.core;
 
+import com.google.common.collect.Lists;
+import uk.gov.register.RegisterConfiguration;
```
Comment (michaelabenyohai) [inline src/main/java/uk/gov/register/core/PostgresRegister.java]: Please can you clean up the imports - this one isn't needed.

File: src/main/java/uk/gov/register/store/postgres/PostgresDriver.java [hunk 2]
```diff
@@ -7,22 +7,21 @@
...
-import java.util.Collection;
-import java.util.List;
-import java.util.Optional;
+import java.util.*;
 
 public abstract class PostgresDriver implements BackingStoreDriver {
 
     protected final MemoizationStore memoizationStore;
+    private final HashMap<String, ArrayList<String>> registerFieldLookup;
+
```
Comment (michaelabenyohai) [inline src/main/java/uk/gov/register/store/postgres/PostgresDriver.java]: This is no longer used. There should be no changes to this file as part of this PR.

File: src/main/java/uk/gov/register/store/postgres/PostgresDriver.java [hunk 3]
```diff
@@ -114,7 +114,7 @@ public int getTotalRecords() {
     }
 
     @Override
-    public List<Record> findMax100RecordsByKeyValue(String key, String value) {
+    public List<Record> findMax100RecordsByKeyValue(String registerName, String key, String value) {
```
Comment (michaelabenyohai) [inline src/main/java/uk/gov/register/store/postgres/PostgresDriver.java]: registerName is not used in this method

File: src/main/java/uk/gov/register/store/postgres/PostgresDriverTransactional.java [hunk 4]
```diff
@@ -7,6 +7,7 @@
 import org.skife.jdbi.v2.exceptions.CallbackFailedException;
 import org.skife.jdbi.v2.tweak.HandleCallback;
 import org.skife.jdbi.v2.tweak.HandleConsumer;
+import uk.gov.register.configuration.RegistersConfiguration;
```
Comment (michaelabenyohai) [inline src/main/java/uk/gov/register/store/postgres/PostgresDriverTransactional.java]: Tidy up imports, this isn't used.

File: src/test/java/uk/gov/register/functional/store/postgres/PostgresDriverTransactionalFunctionalTest.java [hunk 5]
```diff
@@ -17,6 +18,7 @@
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.empty;
 import static org.hamcrest.core.Is.is;
+import static org.mockito.Mockito.mock;
```
Comment (michaelabenyohai) [inline src/test/java/uk/gov/register/functional/store/postgres/PostgresDriverTransactionalFunctionalTest.java]: Tidy up imports

File: src/test/java/uk/gov/register/store/postgres/PostgresDriverTransactionalTest.java [hunk 6]
```diff
@@ -3,10 +3,10 @@
...
 import org.mockito.ArgumentCaptor;
-import uk.gov.register.core.Entry;
-import uk.gov.register.core.Item;
-import uk.gov.register.core.Record;
+import uk.gov.register.configuration.RegistersConfiguration;
+import uk.gov.register.core.*;
 import uk.gov.register.db.*;
+import uk.gov.register.exceptions.NoSuchFieldException;
 import uk.gov.register.views.RegisterProof;
 import uk.gov.verifiablelog.VerifiableLog;
```
Comment (michaelabenyohai) [inline src/test/java/uk/gov/register/store/postgres/PostgresDriverTransactionalTest.java]: Tidy up imports. There should be no change to this file.

File: src/test/java/uk/gov/register/core/PostgresRegisterTest.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+import uk.gov.register.exceptions.NoSuchFieldException;
+import uk.gov.register.store.BackingStoreDriver;
+import uk.gov.register.store.postgres.PostgresDriverTestBase;
+
+import java.util.ArrayList;
+
+import static org.mockito.Mockito.*;
+
+public class PostgresRegisterTest extends PostgresDriverTestBase {
+    private RegisterNameConfiguration registerNameConfiguration;
```
Comment (michaelabenyohai) [inline src/test/java/uk/gov/register/core/PostgresRegisterTest.java]: This does not need to extend PostgresDriverTestBase as these tests do not need the mocked functionality in that class.

Comment (karlbaker02) [inline src/test/java/uk/gov/register/core/PostgresRegisterTest.java]: Good spot, thanks.

**Pull request merged**